hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataAvroResolver.java

```diff
@@ -186,6 +192,28 @@ private static Schema resolveSchema(String recordName, Stream<Field> fields) {
         }).endRecord();
     }
 
+    public static <T> Stream<T> resolveFields(
+            @Nullable Schema schema,
+            String structure,
+            BiFunction<String, QueryDataType, T> constructor
+    ) {
+        if (schema == null) {
+            throw QueryException.error(
```

Comment:
this exception should be thrown higher. This method should not know if it is considered before or after processing explicit fields.